attestation-agent/attestation-agent/src/lib.rs

```diff
@@ -152,17 +197,39 @@ impl AttestationAPIs for AttestationAgent {
         Ok(evidence.into_bytes())
     }
 
-    /// Extend runtime measurement register
+    /// Extend runtime measurement register. Parameters
+    /// - `events`: a event slice. Any single event will be calculated into a hash digest to extend the current
+    /// platform's RTMR.
+    /// - `register_index`: a target PCR that will be used to extend RTMR. Note that different platform
+    /// would have its own strategy to map a PCR index into a architectual RTMR index. If not given, a default one
+    /// will be used.
     async fn extend_runtime_measurement(
         &mut self,
-        events: Vec<Vec<u8>>,
+        domain: &str,
+        operation: &str,
+        content: &str,
         register_index: Option<u64>,
     ) -> Result<()> {
+        let register_index = register_index.unwrap_or_else(|| {
+            info!("No PCR index provided, use default {DEFAULT_PCR_INDEX}");
+            DEFAULT_PCR_INDEX
+        });
+
+        let log_entry = format!("{domain} {operation} {content}");
+        self.eventlog.write_log(&log_entry)?;
```

Comment:
the flow here writes the log and then extends. perhaps a safer approach (following the systemd way) would be to take an flock, extend, write and release.

Reply:
Nice idea. Let me add a Mutex wrapper for `eventlog`